### PR TITLE
Large refactor of sl4a and ed code in android_device. Drop multiple session support.

### PR DESCRIPTION
**Pull request by adorokhine**
Large refactor of sl4a and ed code in android_device.

* Drop multiple session support; it was never used.
* Remove unneeded public methods.
* Rename 'droid' to 'sl4a'

### CODE REVIEW

File: tools/sl4a_client.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    ed (EventDispatcher)
+""".format(serial))
+    gConsoleEnv['ad'].stop_services()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Interactive client for sl4a.')
+    parser.add_argument(
+        '-s', '--serial',
+        help=
```
Comment (xpconanfan) [inline tools/sl4a_client.py]: yapf weirdness?

File: tools/sl4a_client.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    if not serial:
+        serials = android_device.list_adb_devices()
+        if len(serials) != 1:
+            print('ERROR: Expected 1 phone, but %d found. Use the -s flag.' %
+                      len(serials),
+                  file=sys.stderr)
+            sys.exit(1)
+        serial = serials[0]
+    ads = android_device.get_instances([serial])
+    assert len(ads) == 1
```
Comment (xpconanfan) [inline tools/sl4a_client.py]: keeping this assert instead of if...print...exit(1)?

File: mobly/controllers/android_device.py [hunk 3]
```diff
@@ -472,61 +456,6 @@ def model(self):
...
-        there isn't one.
-        """
-        try:
-            session_id = sorted(self._event_dispatchers)[0]
-            return self._event_dispatchers[session_id]
-        except IndexError:
-            return None
-
-    @property
-    def droids(self):
```
Comment (xpconanfan) [inline mobly/controllers/android_device.py]: I'm just happy that these are finally gone +_+

File: tools/sl4a_client.py (hunk 2 above)
Comment (adorokhine) [inline tools/sl4a_client.py]: Don't review sl4a_client here yet, it was forked from that branch prior to your review.
I'll let you know when the sl4a_client PR is submitted and I've rebased this one on top.

File: tools/sl4a_client.py (hunk 1 above)
Comment (adorokhine) [inline tools/sl4a_client.py]: Don't review sl4a_client here yet, it was forked from that branch prior to your review.
I'll let you know when the sl4a_client PR is submitted and I've rebased this one on top.

**Comment by adorokhine** [comment]
Rebased; complete code including sl4a_client now ready for review.

**Comment by xpconanfan** [comment]
LGTM

**Pull request merged**